### PR TITLE
Stats package: Add entrypoint class

### PR DESCRIPTION
**Pull request by fgiannar**
Spin off from: https://github.com/Automattic/jetpack/pull/26348

#### Changes proposed in this Pull Request:

* Adds `Automattic\Jetpack\Stats\Main` entrypoint class for setting hooks and configuration.
* Updates `Automattic\Jetpack\Stats\XMLRPC_Provider` to only add the `jetpack.getBlog` method when the site is connected and the Stats module is active.

#### Other information:

- [x] Have you written new tests for your changes, if applicable?
- [x] Have you checked the E2E test CI results, and verified that your changes do not break them?

#### Jetpack product discussion

p9dueE-5Rd-p2

#### Does this pull request change what data or activity we track or use?

No

#### Testing instructions:

* Look at the code and unit tests

### CODE REVIEW

File: projects/packages/stats/tests/php/test-base.php [hunk 1]
```diff
@@ -81,4 +85,14 @@ protected static function post( $id ) {
...
 		return new WP_Post( $post );
 	}
+
+	/**
+	 * Adds stats to the list of active modules
+	 *
+	 * @param array $modules Array with modules slugs.
+	 * @return array
+	 */
+	public static function filter_jetpack_active_modules( $modules ) {
```
Comment (darssen) [inline projects/packages/stats/tests/php/test-base.php]: This method's name doesn't look representative of what it does.

File: projects/packages/stats/tests/php/test-main.php [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+
+	/**
+	 * Test Main::jetpack_is_dnt_enabled without the `jetpack_honor_dnt_header_for_stats` filter.
+	 */
+	public function test_jetpack_is_dnt_enabled_without_filter() {
+		$_SERVER['HTTP_DNT'] = true;
+
+		$this->assertFalse( Stats::jetpack_is_dnt_enabled() );
+	}
+
```
Comment (darssen) [inline projects/packages/stats/tests/php/test-main.php:109]: For completeness, I would add a test where the filter is added, but the `HTTP_DNT` header is not present, and check that it returns false.
No filter and no header might be too much :)

File: projects/packages/stats/tests/php/test-xmlrpc-provider.php [hunk 3]
```diff
@@ -92,4 +113,14 @@ public function test_get_blog() {
 
 		$this->assertSame( $expected_stats_blog, $stats_blog );
 	}
+
```
Comment (darssen) [inline projects/packages/stats/tests/php/test-xmlrpc-provider.php]: This looks to be duplicated from parent class

**Review by darssen — CHANGES_REQUESTED** [review]
Nice work :)
I have added a couple of minor comments that can be easily addressed.
Besides that, the only thing I encountered is that the `should_track` method could have some more tests.

File: projects/packages/stats/tests/php/test-base.php (hunk 1 above)
Comment (fgiannar) [reply projects/packages/stats/tests/php/test-base.php]: Updated to `filter_jetpack_active_modules_add_stats`

File: projects/packages/stats/tests/php/test-main.php (hunk 2 above)
Comment (fgiannar) [reply projects/packages/stats/tests/php/test-main.php:109]: Added, thanks! :)

File: projects/packages/stats/tests/php/test-xmlrpc-provider.php (hunk 3 above)
Comment (fgiannar) [reply projects/packages/stats/tests/php/test-xmlrpc-provider.php]: Nice catch, removed!

**Comment by fgiannar** [comment]
@darssen Many thanks for reviewing, I have addressed all your inline comments :) 

> Besides that, the only thing I encountered is that the should_track method could have some more tests.

Yeah I know, but I focused on explicitly testing the functionalities we added in this PR vs the whole method (that existed anyway in the Jetpack plugin as is). Adding more tests there would be nice indeed but since this is a blocker for the Stats package work I think it's acceptable for now.

**Comment by darssen** [comment]
Thanks for addressing them so quickly.
I agree with your reasoning regarding the testing so I am approving :)

**Review by darssen — APPROVED** [review]

**Pull request merged**